mne_bids/tests/test_write.py

```diff
@@ -296,12 +296,9 @@ def test_make_dataset_description(tmpdir):
         dataset_description_json = json.load(fid)
         assert dataset_description_json["Authors"] == ['MNE B.', 'MNE P.']
 
+    monkeypatch.setattr(write, 'BIDS_VERSION', 'old')
     with pytest.raises(ValueError, match='Previous BIDS version used'):
-        version = make_dataset_description.__globals__['BIDS_VERSION']
-        make_dataset_description.__globals__['BIDS_VERSION'] = 'old'
         make_dataset_description(path=tmpdir, name='tst')
-        # put version back so that it doesn't cause issues down the road
-        make_dataset_description.__globals__['BIDS_VERSION'] = version
```

Comment:
I find this much simpler and it solves the problem without any need for `__globals__` / waiting for 0.24 / https://github.com/mne-tools/mne-python/pull/9732, @alexrockhill see if you agree...

Reply:
BTW I'm not sure how these lines were ever hit, once the exception is raised I don't think the lines that follow are executed:
```
with pytest.raises(RuntimeError):
    raise RuntimeError
    print('foo')
```
does not print anything...

Reply:
Nice, looks great